### PR TITLE
chore: adding optional pre-commit hooks [MTT-3302]

### PR DESCRIPTION
**Pull request by SamuelBellomo**
### Description

Adding pre-commit hooks to help keep bad things away. These will fail your commits if you're trying to include project IDs or black listed packages.
To install those, run the install script from the RepoUtilities folder.
Not tested on Windows yet.

### Issue Number(s)

[MTT-3302](https://jira.unity3d.com/browse/MTT-3302)

### Contribution checklist
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] JIRA ticket ID is in the PR title or at least one commit message
 - [ ] Include the ticket ID number within the body message of the PR to create a hyperlink

### CODE REVIEW

**Comment by SamuelBellomo** [comment]
This is part 1 of a series of PRs

**Review by bendoyon — APPROVED** [review]

File: RepoUtilities/pre-commit.toInstall [hunk 1]
```diff
@@ -0,0 +1 @@
+python3 RepoUtilities/pre-commit.py
```
Comment (LPLafontaineB) [inline RepoUtilities/pre-commit.toInstall:1]: On Windows, it did not work for me, and prevented me from doing any commits, with this error: 
error: cannot spawn .git/hooks/pre-commit: No such file or directory

Adding a shebang like #!/bin/sh at the top of the pre-commit file ended up solving the issue. If it's something that works on mac as well, let's add it here.

Comment (fernando-cortez) [reply RepoUtilities/pre-commit.toInstall:1]: Seconding that for Windows.

Comment (SamuelBellomo) [reply RepoUtilities/pre-commit.toInstall:2]: pushed the shebang, let me know if there's other issues

File: RepoUtilities/pre-commit.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    cmd = f"git cat-file blob :{filePath} | git lfs smudge"
+    stream = os.popen(cmd)
+    return stream.read()
+
+output = GetStagedFileContent('Packages/manifest.json')
+packages = json.loads(output)['dependencies']
+if ('com.unity.multiplayer.virtualprojects' in packages):
+    FailCommit("Virtual projects in packages")
+if ('github' in packages['com.unity.multiplayer.tools'].lower()):
+    FailCommit("Tools using github package")
```
Comment (becksebenius-unity) [inline RepoUtilities/pre-commit.py:25]: Is there an easy way to bypass this? Tools has public releases and there will be valid commits periodically to reference new versions

Comment (becksebenius-unity) [reply RepoUtilities/pre-commit.py:25]: OH - I misread this line, I didn't see the `'github'` part. This is fine!

**Review by becksebenius-unity — APPROVED** [review]

**Review by LPLafontaineB — APPROVED** [review]

**Pull request merged**